frontend/packages/pipelines-plugin/src/components/quicksearch/PipelineQuickSearchVersionDropdown.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import { Dropdown, DropdownItem, DropdownToggle } from '@patternfly/react-core';
+import { CheckCircleIcon } from '@patternfly/react-icons';
+import { global_palette_green_500 as greenColor } from '@patternfly/react-tokens/dist/js/global_palette_green_500';
+import { CatalogItem } from 'packages/console-dynamic-plugin-sdk/src';
+import { isSelectedVersionInstalled } from './pipeline-quicksearch-utils';
+
+interface PipelineQuickSearchVersionDropdownProps {
+  selectedVersion: string;
+  item: CatalogItem;
+  onChange: (key: string) => void;
+}
+
+const PipelineQuickSearchVersionDropdown: React.FC<PipelineQuickSearchVersionDropdownProps> = ({
+  item,
+  onChange,
+  selectedVersion,
+}) => {
+  const [isOpen, setOpen] = React.useState(false);
+  const toggleIsOpen = React.useCallback(() => setOpen((v) => !v), []);
+  const versions = item?.attributes?.versions ?? [];
+  const versionItems = versions.reduce((acc, { version, id }) => {
+    acc[id.toString()] = id === item.data?.latestVersion?.id ? `${version} (latest)` : version;
+    return acc;
+  }, {});
+
+  const getDropdownItem = React.useCallback(
+    (key: string): React.ReactNode => (
+      <DropdownItem
+        component="button"
+        key={key}
+        label={versionItems[key]}
+        onClick={(e) => {
+          e.stopPropagation();
+          onChange(key);
+          setOpen(false);
+        }}
+      >
+        <div className="opp-quick-search-details__version-dropdown-item">
+          {versionItems[key]}
+          {isSelectedVersionInstalled(item, key) && <CheckCircleIcon color={greenColor.value} />}
+        </div>
+      </DropdownItem>
+    ),
+    [item, onChange, versionItems],
+  );
+
+  return (
+    <Dropdown
+      data-test={'task-version'}
+      className="opp-quick-search-details__version-dropdown"
+      dropdownItems={Object.keys(versionItems).map(getDropdownItem)}
```

Comment:
Hmmm, a little curious you took an array to an object to essentially just return it back to an array. The best I can see is you did this for the `selectedVersion` one. It is a little wonky to read. Perhaps just go straight to `DropdownItems`? You can get the selectedVersion through a find I would think.

I guess this mainly a Nit as it doesn't really change the way or really overly improve things. Just was odd following the flow reading the code.

Reply:
versionItems here is an object to array of dropDownItems. I moved it to dropdownItems.